regression: stdout get unexpected lines with traces pointing to PMI code.

```
======================================================================
FAIL: testSuccess (__main__.TestRunCommand.testSuccess)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/mpi4py-testing/mpi4py-testing/mpi4py/demo/test-run/test_run.py", line 246, in testSuccess
    self.assertEqual(stderr, '')
AssertionError: '[0] 680 at [0x55a35136b7e0], src/util/mpir_pmi.c[303]\n' != ''
- [0] 680 at [0x55a35136b7e0], src/util/mpir_pmi.c[303]
```
Full logs can be found [here](https://github.com/mpi4py/mpi4py-testing/actions/runs/5549556381/jobs/10137798614).
Are those memory leaks? 
Maybe a regression from #6580?

PS: I'm surprised MPICH's own test suite did not catch this issue before. Could it be that the issue is actually on mpi4py side?

--- a/src/util/mpir_hwtopo.c
+++ b/src/util/mpir_hwtopo.c
@@ -210,6 +210,7 @@ int MPII_hwtopo_init(void)
              * file is really the underlying system. */
             hwloc_topology_set_flags(hwloc_topology, HWLOC_TOPOLOGY_FLAG_IS_THISSYSTEM);
         }
+        MPL_free(xmlfile);
     }
 
     hwloc_topology_set_io_types_filter(hwloc_topology, HWLOC_TYPE_FILTER_KEEP_ALL);

